### PR TITLE
exclude developer tools from go modules.

### PR DESCRIPTION
**Pull request by pei0804**
**Describe the PR**
swag is't need `go lint` for go modules.

### CODE REVIEW

**Comment by pei0804** [comment]
I think about a little more.

File: Makefile [hunk 1]
```diff
@@ -53,7 +53,7 @@ install:
 
 .PHONY: lint
 lint:
-	which golint || $(GOGET) -u golang.org/x/lint/golint
+	which golint || GO111MODULE=off $(GOGET) -v -u golang.org/x/lint/golint
```
Comment (easonlin404) [inline Makefile]: why should set `GO111MODULE=off` on this?  😕

Comment (pei0804) [reply Makefile]: If you don't turn it off temporarily, dependencies will be added to go mod.

```
❯ make lint
go: finding github.com/KyleBanks/depth v1.2.1
go: finding github.com/ghodss/yaml v1.0.0
go: finding github.com/swaggo/gin-swagger v1.2.0
go: finding github.com/swaggo/files v0.0.0-20190704085106-630677cd5c14
go: finding github.com/stretchr/testify v1.3.0
go: finding github.com/alecthomas/template v0.0.0-20190718012654-fb15b899a751
go: finding github.com/go-openapi/spec v0.19.0
go: finding github.com/go-openapi/jsonreference v0.19.0
go: finding github.com/urfave/cli v1.20.0
go: finding golang.org/x/tools v0.0.0-20190611222205-d73e1c7e250b
go: finding github.com/stretchr/testify v1.2.2
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/go-openapi/swag v0.17.0
go: finding github.com/go-openapi/jsonpointer v0.17.0
go: finding github.com/mailru/easyjson v0.0.0-20180823135443-60711f1a8329
go: finding gopkg.in/yaml.v2 v2.2.1
```

File: Makefile [hunk 2]
```diff
@@ -45,19 +51,25 @@ clean:
...
-	$(GOGET) -v github.com/swaggo/swag github.com/swaggo/swag/cmd/swag github.com/swaggo/swag/gen
-	$(GOGET) github.com/alecthomas/template
+.PHONY: deps
+deps:
+	$(GOGET) ${u} -d
 	$(GOGET) github.com/stretchr/testify/assert
+	$(GOGET) github.com/alecthomas/template
+
+.PHONY: dev-tool
+dev-tool:
```
Comment (pei0804) [inline Makefile]: Just install the dev tool.

**Comment by pei0804** [comment]
@easonlin404 
plz review.

File: Makefile [hunk 3]
```diff
@@ -45,19 +51,25 @@ clean:
...
-install:
-	$(GOGET) -v github.com/swaggo/swag github.com/swaggo/swag/cmd/swag github.com/swaggo/swag/gen
-	$(GOGET) github.com/alecthomas/template
+.PHONY: deps
+deps:
+	$(GOGET) ${u} -d
 	$(GOGET) github.com/stretchr/testify/assert
+	$(GOGET) github.com/alecthomas/template
+
+.PHONY: devel-deps
```
Comment (pei0804) [inline Makefile]: Just install the dev tool.

File: Makefile [hunk 4]
```diff
@@ -3,11 +3,13 @@ GOLINT:=$(shell which golint)
 GOIMPORT:=$(shell which goimports)
 GOFMT:=$(shell which gofmt)
 GOBUILD:=$(GOCMD) build
+GOINSTALL:=$(GOCMD) install
 GOCLEAN:=$(GOCMD) clean
 GOTEST:=$(GOCMD) test
 GOGET:=$(GOCMD) get
 GOLIST:=$(GOCMD) list
 GOVET:=$(GOCMD) vet
+u := $(if $(update),-u)
```
Comment (pei0804) [inline Makefile]: if want to update.
`update=-u; make deps`

**Pull request merged**